src/renderers/dom/shared/__tests__/ReactDOMComponent-test.js

```diff
@@ -260,6 +260,75 @@ describe('ReactDOMComponent', function() {
       expect(container.firstChild.hasAttribute('height')).toBe(false);
     });
 
+    it('should remove known SVG camel case attributes', function() {
+      var container = document.createElement('div');
+      ReactDOM.render(<svg viewBox="0 0 100 100" />, container);
+
+      expect(container.firstChild.hasAttribute('viewBox')).toBe(true);
+      ReactDOM.render(<svg />, container);
+      expect(container.firstChild.hasAttribute('viewBox')).toBe(false);
+    });
+
+    it('should remove known SVG hyphenated attributes', function() {
+      var container = document.createElement('div');
+      ReactDOM.render(<svg clip-path="0 0 100 100" />, container);
+
+      expect(container.firstChild.hasAttribute('clip-path')).toBe(true);
+      ReactDOM.render(<svg />, container);
+      expect(container.firstChild.hasAttribute('clip-path')).toBe(false);
+    });
+
+    it('should remove arbitrary SVG hyphenated attributes', function() {
+      var container = document.createElement('div');
+      ReactDOM.render(<svg the-word="the-bird" />, container);
+
+      expect(container.firstChild.hasAttribute('the-word')).toBe(true);
+      ReactDOM.render(<svg />, container);
+      expect(container.firstChild.hasAttribute('the-word')).toBe(false);
+    });
+
+    it('should remove arbitrary SVG camel case attributes', function() {
+      var container = document.createElement('div');
+      ReactDOM.render(<svg theWord="theBird" />, container);
+
+      expect(container.firstChild.hasAttribute('theWord')).toBe(true);
+      ReactDOM.render(<svg />, container);
+      expect(container.firstChild.hasAttribute('theWord')).toBe(false);
+    });
+
+    it('should remove SVG attributes that should have been hyphenated', function() {
+      spyOn(console, 'error');
+      var container = document.createElement('div');
+      ReactDOM.render(<svg clipPath="0 0 100 100" />, container);
+      expect(console.error.argsForCall.length).toBe(1);
+      expect(console.error.argsForCall[0][0]).toContain('clipPath');
+      expect(console.error.argsForCall[0][0]).toContain('clip-path');
+
+      expect(container.firstChild.hasAttribute('clip-path')).toBe(true);
+      ReactDOM.render(<svg />, container);
+      expect(container.firstChild.hasAttribute('clip-path')).toBe(false);
+    });
+
+    it('should remove namespaced SVG attributes', function() {
+      var container = document.createElement('div');
+      ReactDOM.render(
+        <svg>
+          <image xlinkHref="http://i.imgur.com/w7GCRPb.png" />
```

Comment:
fun story - at Mozilla we had tests randomly start breaking one day and it took us a while to figure out that it was because we had a test trying to load a remote image which no longer existed. I don't expect that to happen here but I wanted to call it out just in case I'm long gone and it does happen. Also, we already have a test with this exact svg image so meh.